### PR TITLE
Body.inBody should return true when element is within a shadow root within the body

### PR DESCRIPTION
**Pull request by tinydylan**
Related Ticket: TINY-6076

Description of Changes:
* Placeholder text

Pre-checks:
* [x] Changelog entry added
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)
* [X] License headers added on new files (if applicable)

Review:
* [X] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Comment by tinydylan** [comment]
Setting to draft until build failures are fixed.

**Comment by tinydylan** [comment]
Is a changelog required for this PR?

File: modules/sugar/src/test/ts/browser/BodyTest.ts [hunk 1]
```diff
@@ -16,12 +17,65 @@ UnitTest.test('BodyTest', () => {
...
+  const body = SelectorFind.first('body').getOrDie();
 
-  Remove.remove(div);
+  const div = Element.fromTag('div');
+  const child = Element.fromTag('span');
+  const text = Element.fromText('hi');
+  Insert.append(child, text);
+  Insert.append(div, child);
+  Insert.append(body, div);
+  try {
```
Comment (lnewson) [inline modules/sugar/src/test/ts/browser/BodyTest.ts:31]: Another place where we maybe shouldn't have a try/finally?

**Comment by lnewson** [comment]
> Is a changelog required for this PR?

I'd say no, so I've marked that off the list.

File: modules/sugar/src/test/ts/browser/BodyTest.ts (hunk 1 above)
Comment (tinydylan) [reply modules/sugar/src/test/ts/browser/BodyTest.ts]: Ta. Fixed.

**Pull request merged**